### DISCUSSION TITLE
editable tab title switches to preview mode on space press

### DISCUSSION THREAD

**Discussion by mwmcode** [chakra-ui/chakra-ui#2557] [General]
Hello,

I'm trying to have an editable `<Tab />` title using `<Editable />` component

[Live example on Codesandbox](https://codesandbox.io/s/mutable-glitter-s8r1d?file=/src/App.tsx) 

Snippet
```jsx
          <Tab >
            <Editable defaultValue="title">
              <EditablePreview />
              <EditableInput />
            </Editable>
          </Tab>
```

I'm facing an issue [due to `<Tab />` handling Space key press](https://chakra-ui.com/docs/components/tabs#tabs-props)  where once the space bar is hit/registered, the editable component goes back to `preview` mode

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/13101565/99496219-faed0480-29c7-11eb-8b16-fd9446146553.gif)

Is there a workaround this? 

PS: congrats on reaching `v1` milestone 🎉

**Comment by FdelMazo** [comment]
Did you find any workaround @mwmcode ?

**Comment by FdelMazo** [comment]
Hi @mwmcode, I know it would be ideal for chakra to implement a prop for this, but at least I found a workaround that's working for my usecase:

- Create a React ref for the `EditableInput` component
- Prevent the default action on space presses, on the `Tabs` component
- Add to `ref.current.value` a space, manually, where the user is placed

```
const inputref = React.useRef(null)
...
      <Tabs onKeyDown={(e) => {
            if (e.keyCode === 32) {
                        e.preventDefault()
                        const ch = e.target.selectionStart;
                        inputref.current.value = e.target.value.slice(0, ch) + " " + e.target.value.slice(ch);
                        e.target.selectionStart = ch + 1
                        e.target.selectionEnd = ch + 1
            }
          }}>
          <Tab>
            <Editable defaultValue="title">
              <EditablePreview />
              <EditableInput ref={inputref}/>
            </Editable>
          </Tab>
      </Tabs>
```

**Reply by mwmcode** [reply]
Hi @FdelMazo sorry I missed your earlier comment, I didn't end up implementing this after all.

Your solution may work but _I think_ it means sacrificing an important accessibility feature: pressing space activates/opens tab.

Maybe if you can check if the user is editing a tab's editable component then `e.preventDefault()` otherwise: let the event propagate so the `<Tabs />` component acts as expected.

**Reply by FdelMazo** [reply]
> sacrificing an important accessibility feature

You are totally right, I missed that!
